src/flags/fsFlags.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import * as fs from 'fs';
+import { Flags } from '@oclif/core';
+import { Messages } from '@salesforce/core';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/sf-plugins-core', 'messages');
+
+/**
+ * Accepts a directory path.  Validates that the directory exists and is a directory.
+ */
+export const existingDirectory = Flags.build<string>({
```

Comment:
@mshanemc would it be possible to collapse `existingDirectory` and `existingFile` into one flag, allowing one to choose as part of the flag definition which should be checked for dir or filepath

Reply:
Possible, but then you have to expose that more complex buildX to the consumer and have parameters.  And what would you name that combined flag?